enc_bootloader.h

```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright (c) 2020 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#pragma once
+
+#include <memory>
+#include <fstream>
+
+std::shared_ptr<std::iostream> get_enc_bootloader(bool use_mbedtls = false);
```

Comment:
This function doesn't seem t be used anywhere? :confused:

Reply:
It is, in `encrypt_command` in main.cpp

Reply:
DOH! Github was showing me the "Large diffs are not rendered by default." for `main.cpp` when I initially tried searching for where this function was used :man_facepalming: 

_However_ I wonder if it might be worth renaming `enc_bootloader.cpp` and `enc_bootloader.h` to something else, in order to avoid any possible confusion with the `enc_bootloader.c` that's also in this PR? :thinking:  (AFAICT the former is code that gets run on your PC, and the latter is code that gets run on your Pico?)

Reply:
I've renamed `enc_bootloader.cpp/.h` to `get_enc_bootloader.cpp/.h`, and done the same for `xip_ram_perms.cpp`, as that's more what the cpp files are doing (they just locate the ELFs to run on the Pico)